ios/ReactNativeCameraKit/RealCamera.swift

```diff
@@ -244,36 +253,33 @@ class RealCamera: NSObject, CameraProtocol, AVCaptureMetadataOutputObjectsDelega
             self.update(torchMode: self.torchMode)
         }
     }
-
+    
+    func counterRotatedCaptureVideoOrientationFrom(deviceOrientation: UIInterfaceOrientation) -> AVCaptureVideoOrientation? {
```

Comment:
I was very tired and had to finish up. It was because UI and hardware orientation are opposed to each other in landscape mode (device is rotated left, so UI must rotate right to be readable).
The switch was indeed counter rotating until the end.
I think it's because `initializeMotionManager` is already counter-rotating the device orientation, which is a bug.

So it may be worth checking if we should:
1. Keep the name the same
2. Change `initializeMotionManager` so it doesn't counter-rotate
3. Change the initial value of `self.deviceOrientation` when it's initialized to the UI orientation such that it also counter-rotates back (so if UI is in landscape-left, device orientation must be assumed to be landscape-right).

Reply:
Gotcha, I'm currently looking at 2.

Reply:
@scarlac I refined the orientation implementation. Let me know what you think - feel free to revert if you preferred yours

Every time `UIDeviceOrientation` is converted to `AVCaptureVideoOrientation`, we counter rotate it
It applies to
- Accelerometer, which is now setting `deviceOrientation` (`UIDeviceOrientation`), not counter rotated
- `uiOrientationChanged`, `device.orientation`

Reply:
`UIDeviceOrientation` and `...AVCAptureVideoOrientation` are not the same thing. They are conceptually different and so you cannot use them as the same value.
When you orient the device left, the UI must rotate right. The counter-rotation function is necessary.
Can we undo those changes? The code is necessary. You cannot simplify it. We must have the actual device orientation, both for onOrientationChange() events and for selectively updating the video stream orientation. Both are needed, separately, for separate purposes.

Reply:
Yes, I think we are aligned.

Current code does counter rotate when converting `device` to `interface/video`. End result is exact same as yours

What's misleading is the name `UIDeviceOrientation` which contains `UI` & `Device`

- `UIDeviceOrientation` is physical / device
- `UIInterfaceOrientation ` is interface
- `AVCAptureVideoOrientation` is video, synced with interface

Converting `UIDeviceOrientation` to `UIInterfaceOrientation`/`AVCAptureVideoOrientation` requires counter rotation left/right.
That counter rotation is abstracted in the conversion extension, so as a consumer they don't have to know about those subtleties
- Asking for `.videoOrientation` from a `UIInterfaceOrientation` IS NOT counter rotated
- Asking for `.videoOrientation` from a `UIDeviceOrientation` IS counter rotated

I'm fine using methods, it's mainly code style. My only advice is to code it so we could add unit tests later, as it is easy to break